plugins/in_podman_metrics/podman_metrics.h

```diff
@@ -0,0 +1,88 @@
+/* -*- Mode: C; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- */
+
+/*  Fluent Bit
+ *  ==========
+ *  Copyright (C) 2015-2022 The Fluent Bit Authors
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+#ifndef FLB_IN_PODMAN_METRICS_H
+#define FLB_IN_PODMAN_METRICS_H
+
+#include <fluent-bit/flb_info.h>
+#include <fluent-bit/flb_config.h>
+#include <fluent-bit/flb_input.h>
+#include <fluent-bit/flb_utils.h>
+#include <fluent-bit/flb_jsmn.h>
+
+#include <monkey/mk_core/mk_list.h>
+
+#include "podman_metrics_config.h"
+
+static int collect_container_data(struct flb_in_metrics *ctx);
+static int add_container_to_list(struct flb_in_metrics *ctx, flb_sds_t id, flb_sds_t name);
+static int destroy_container_list(struct flb_in_metrics *ctx);
+static int destroy_counters(struct flb_in_metrics *ctx);
+
+static int create_counter(struct flb_in_metrics *ctx, struct cmt_counter **counter, flb_sds_t id, flb_sds_t name, flb_sds_t metric_prefix,
+                          flb_sds_t *fieds, flb_sds_t metric_name, flb_sds_t description, flb_sds_t interface, uint64_t value);
+static int create_counters(struct flb_in_metrics *ctx);
+
+static int scrape_metrics(struct flb_config *config, struct flb_in_metrics *ctx);
+
+static int cb_metrics_collect_runtime(struct flb_input_instance *ins, struct flb_config *config, void *in_context);
+static int in_metrics_init(struct flb_input_instance *in, struct flb_config *config, void *data);
+static int in_metrics_exit(void *data, struct flb_config *config);
+static void in_metrics_pause(void *data, struct flb_config *config);
+static void in_metrics_resume(void *data, struct flb_config *config);
+
+
+static struct flb_config_map config_map[] = {
+    {
+     FLB_CONFIG_MAP_TIME, "scrape_interval", "20",
```

Comment:
Any reason for this default? 

I had a quick look and it appears we have scrape_interval defaults currently of 1,2,5,10 and 30 seconds depending on the plugin used. Personally I would prefer we try to keep to one of the existing ones (e.g. 10s) and reduce this sprawl but it is only a nit pick.

Reply:
I'm having trouble to remember why I picked this particular value - I've changed it to 30s

Reply:
As for unit tests: I've checked it when i started to write this plugin, but it seems there are no tests (so no framework, no functions etc) for metric related plugins. Do you think it is still worth the effort? I can unit test particular mechanisms without checking counter values from "outside"

Reply:
Right, I've added unit tests for my plugin - they generate metrics from various sources (or their absence). Since it is metric plugin and it doesn't publish data like "normal ones", I had to dig into plugin internals in my tests (in order to get harvested metrics).

Attaching log file from unit test execution:
[unit_tests.log](https://github.com/fluent/fluent-bit/files/9541440/unit_tests.log)

Reply:
Thanks @pabloxxl